test/e2e/cypress/integration/profiles/code/registration/error.spec.ts

```diff
@@ -75,12 +77,14 @@ context("Registration error messages with code method", () => {
 
         cy.get('input[name="traits.email"]').type(email)
         cy.submitCodeForm()
-
-        cy.url().should("contain", "registration")
+        cy.get('[data-testid="ui/message/1040005"]').should(
```

Comment:
@Benehiko the most important aspect of e2e tests are wait assertions. Before:

1. Go to `/registration`
2. Submit form
3. Wait for URL to be `registration` (it already is registration though!)
4. Flaking test failure because sometimes the wait condition is done before the submit is done

Instead, use proper wait conditions that wait for e.g. specific elements such as this one.